src/plugins/section_category.ts

```diff
@@ -0,0 +1,257 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Model, SectionCategory } from '@manuscripts/json-schema'
+import { ManuscriptNode, schema } from '@manuscripts/transform'
+import { EditorState, Plugin, PluginKey } from 'prosemirror-state'
+import { Decoration, DecorationSet, EditorView } from 'prosemirror-view'
+
+import { sectionCategoryIcon } from '../assets'
+import { EditorProps } from '../configs/ManuscriptsEditor'
+import { PopperManager } from '../lib/popper'
+import {
+  getCategoryName,
+  isBackMatterSection,
+  isEditableSectionCategory,
+  isUnique,
+} from '../lib/section-categories'
+import { isChildOfNodeType } from '../lib/utils'
+
+export const sectionCategoryKey = new PluginKey<PluginState>('section-category')
+const popper = new PopperManager()
+
+export interface PluginState {
+  decorations: DecorationSet
+}
+
+export interface SectionCategoryProps extends EditorProps {
+  sectionCategories: SectionCategory[]
+}
+
+function changeNodeType(
+  view: EditorView,
+  node: ManuscriptNode,
+  type: string,
+  pos: number
+) {
+  view.dispatch(
+    view.state.tr.setNodeMarkup(pos, undefined, {
+      ...node.attrs,
+      category: type,
+    })
+  )
+}
+
+function createMenuItem(
```

Comment:
all the functions that are not defined inside the plugin context can be moved int a separate file as a part of this plugin. A folder for a plugin can be created as we did for footnotes and some other plugins.

Reply:
done